tests/block_test.go

```diff
@@ -61,6 +65,85 @@ func TestBlockchain(t *testing.T) {
 	// which run natively, so there's no reason to run them here.
 }
 
+func networkPostMerge(network string) bool {
+	switch network {
+	case "Frontier":
+		return false
+	case "EIP150":
+		return false
+	case "EIP158":
+		return false
+	case "Byzantium":
+		return false
+	case "Constantinople":
+		return false
+	case "ConstantinopleFix":
+		return false
+	case "Istanbul":
+		return false
+	case "MuirGlacier":
+		return false
+	case "Berlin":
+		return false
+	case "London":
+		return false
+	case "ArrowGlacier":
+		return false
+	case "GreyGlacier":
+		return false
+	case "Merge":
+		return true
+	case "Shanghai":
+		return true
+	case "Cancun":
+		return true
+	}
+	return false
+}
+
+func TestStatelessBlockchain(t *testing.T) {
+	bt := new(testMatcher)
+	// General state tests are 'exported' as blockchain tests, but we can run them natively.
+	// For speedier CI-runs, the line below can be uncommented, so those are skipped.
+	// For now, in hardfork-times (Berlin), we run the tests both as StateTests and
+	// as blockchain tests, since the latter also covers things like receipt root
+	bt.skipLoad(`^GeneralStateTests/`)
+
+	// Skip random failures due to selfish mining test
+	bt.skipLoad(`.*bcForgedTest/bcForkUncle\.json`)
+
+	// Slow tests
+	bt.slow(`.*bcExploitTest/DelegateCallSpam.json`)
+	bt.slow(`.*bcExploitTest/ShanghaiLove.json`)
+	bt.slow(`.*bcExploitTest/SuicideIssue.json`)
+	bt.slow(`.*/bcForkStressTest/`)
+	bt.slow(`.*/bcGasPricerTest/RPC_API_Test.json`)
+	bt.slow(`.*/bcWalletTest/`)
+
+	// Very slow test
+	bt.skipLoad(`.*/stTimeConsuming/.*`)
+	// test takes a lot for time and goes easily OOM because of sha3 calculation on a huge range,
+	// using 4.6 TGas
+	bt.skipLoad(`.*randomStatetest94.json.*`)
+
+	// skip uncle tests for stateless
+	bt.skipLoad(`.*/UnclePopulation.json`)
+
+	bt.walk(t, blockTestDir, func(t *testing.T, name string, test *BlockTest) {
+		if runtime.GOARCH == "386" && runtime.GOOS == "windows" && rand.Int63()%2 == 0 {
+			t.Skip("test (randomly) skipped on 32-bit windows")
+		}
+		if networkPostMerge(test.json.Network) {
```

Comment:
Actually, you could remove that `networkPostMerge` with something like this instead: 
```golang
		config, ok := Forks[t.json.Network]
		if !ok {
			return UnsupportedForkError{t.json.Network}
		}
		isMerged := config.TerminalTotalDifficulty != nil && config.TerminalTotalDifficulty.BitLen() == 0
		if !isMerged {
```